src/test/expression/expression_evaluator_test.cpp

```diff
@@ -441,13 +441,18 @@ TEST_F(ExpressionEvaluatorTest, Parameter) {
 TEST_F(ExpressionEvaluatorTest, InListLiterals) {
   EXPECT_TRUE(test_expression<int32_t>(*in_(null_(), list_(null_())), {std::nullopt}));
   EXPECT_TRUE(test_expression<int32_t>(*in_(null_(), list_(null_(), 3)), {std::nullopt}));
-  EXPECT_TRUE(test_expression<int32_t>(*in_(null_(), list_()), {0}));
+  EXPECT_TRUE(test_expression<int32_t>(*in_(null_(), list_()), {std::nullopt}));
```

Comment:
why did the semantics of `NULL IN ()` change? In my experience, an expression involving NULL results in NULL IFF the expression result changes if the NULL would be replaced with a different value. 
That is why `NULL AND FALSE` is FALSE, but `NULL OR FALSE` is NULL. In the case of `NULL in ()` the result should be FALSE since whatever you replace NULL with, the result of the expression is always FALSE.

Reply:
Makes sense.